subsys/bluetooth/mesh/access.c

```diff
@@ -1506,8 +1506,10 @@ int bt_mesh_model_send(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
 		       struct net_buf_simple *msg,
 		       const struct bt_mesh_send_cb *cb, void *cb_data)
 {
-	if (IS_ENABLED(CONFIG_BT_MESH_OP_AGG) && bt_mesh_op_agg_accept(ctx)) {
-		return bt_mesh_op_agg_send(model, ctx, msg, cb);
+	if (IS_ENABLED(CONFIG_BT_MESH_OP_AGG_SRV) && bt_mesh_op_agg_srv_accept(ctx, msg)) {
```

Comment:
It just made most sense to me. The aggregation process for the server is fully dependent on internal behavoir and should under normal circumstances complete in a very short time. The client aggregation however is completly dependent on how the user utilize the API. I wanted to avoid a situation where starting a client sequence on the device would block any incoming sequences to the server, since doing it the other way around would put the status responses in the ongoing aggregated sequence buffer rather than in the response buffer. Ideally (as discussed outside this PR) we should have a way to tag access messages to know if the message belongs to a agg sequence or a agg response (or non of them). Then we could remove the `else if` and make the decision based on the tag. But this would require major rework and potentially change API, so I think this is the best solution for now if we don't want to spend much more time fixing this properly.

Reply:
Should we create a separate task for this?

Reply:
Added to NCSDK-23176